controller/user-controller.js

```diff
@@ -0,0 +1,146 @@
+const db = require("../models");
+const config = require("../config/auth.config");
+
+var jwt = require("jsonwebtoken");
+var bcrypt = require("bcryptjs");
+
+exports.signup = (req, res) => {
+  // Generate account No
+  const accountNo = () => {
+    let num = "";
+    while (num.length < 10) {
+      num += Math.floor(Math.random() * 10);
+    }
+    return num;
+  };
+  const generateAccountNo = accountNo();
+
+  //  Generate Otp
+  const generateOtp = () => {
+    let num = "";
+    while (num.length < 4) {
+      num += Math.floor(Math.random() * 4);
+    }
+    return num;
+  };
+  const generateNewOtp = generateOtp();
+
+  // Save User to Database
+  db.User.create({
+    fullname: req.body.fullname,
+    username: req.body.username,
+    email: req.body.email,
+    password: bcrypt.hashSync(req.body.password, bcrypt.genSaltSync(8), null),
+    accountNumber: generateAccountNo,
+    phoneNumber: req.body.phoneNumber,
+    otp: generateNewOtp,
+  })
+    .then((user) => {
+      res.status(201).json({
+        status: "ok",
+        message: "User registered",
+        data: {
```

Comment:
Why are you not returning everything but the password and otp? What happens in the case when we want to integrate welcome emails?

Reply:
I have returned everything needed